pennylane/pauli/__init__.py

```diff
@@ -48,6 +48,7 @@
     PauliGroupingStrategy,
     optimize_measurements,
     graph_colouring,
+    compute_partition_indices,
```

Comment:
A routine problem in pennylane is exposing things to other modules for PL developers to use, without necessarily exposing it to users.  Basically there is no way to differentiate the two, and we end up expecting users to rely on anything we have defined anywhere.

But since we are using it other modules, we should probably have it available as `qml.pauli.compute_partition_indices`.

Reply:
Thanks @albi3ro! I'm generally in favour of developing "deep classes" (with minimal APIs), but it doesn't mean that the current implementation conveys a "shallow class". Not a blocker from my side to continue with the current solution.